Define and activate kvm 12.1.0 release (#376)
giantswarm/releases: kvm/v12.1.0/README.md

## :zap:  Giant Swarm Release 12.1.0 for KVM :zap:

> **This release includes two significant improvements to NGINX Ingress Controller. It also adds support for image registry mirrors, improving reliability.**

Reviewer: ```suggestion
**This release includes two significant improvements to NGINX Ingress Controller. It also includes a fix for Quay being a single point of failure by using Docker mirroring feature.
This ensures availability of all images needed for node bootstrap, thus the cluster creation / scaling doesn’t depend on Quay availability anymore.**
```

## :zap:  Giant Swarm Release 12.1.0 for KVM :zap:

> In this release:

Reviewer 2: ```suggestion
**This release includes two significant improvements to NGINX Ingress Controller. It also adds support for image registry mirrors, improving reliability.**
```

Just to give a top level summary.

Reviewer 2: @sslavic @snizhana-dynnyk I think the registry mirrors is a pretty significant change and we should call out in the main paragraph. Is this wrong? :D

Reviewer 2: Should we use the language used in the AWS release? https://github.com/giantswarm/releases/tree/master/aws/v11.5.0

Reviewer: yes, but without mentioning China

> - Multiple NGINX Ingress Controllers per tenant cluster are supported.

Reviewer 2: ```suggestion
**The two NGINX Ingress Controller improvements:**
* First, multiple NGINX Ingress Controllers per tenant cluster are now supported, enabling separation of internal vs external traffic, dev vs prod, and so on.
* Second, management of NGINX IC NodePort Service is moved from kvm-operator to NGINX IC App itself. This enables configurability of external traffic policy and lays the foundation for making NGINX IC App optional and not pre-installed in a future KVM platform release.
```

## :zap:  Giant Swarm Release 12.1.0 for KVM :zap:

**This release includes two significant improvements to NGINX Ingress Controller. It also adds support for image registry mirrors, improving reliability.**

**The two NGINX Ingress Controller improvements:**
* First, multiple NGINX Ingress Controllers per tenant cluster are now supported, enabling separation of internal vs external traffic, dev vs prod, and so on.
* Second, management of NGINX IC NodePort Service is moved from kvm-operator to NGINX IC App itself. This enables configurability of external traffic policy and lays the foundation for making NGINX IC App optional and not pre-installed in a future KVM platform release.

> - NGINX IC NodePort Service external traffic policy has been made configurable.

Reviewer 2: ```suggestion
```

> - Management of NGINX IC NodePort Service is moved from kvm-operator to NGINX IC App itself to:

Reviewer 2: ```suggestion
```

>   - Enable external traffic policy configurability, in a way consistent with other NGINX IC configuration options.

Reviewer 2: ```suggestion
```

>   - Lay the foundation for making NGINX IC App optional and not pre-installed in a future KVM platform release.

Reviewer 2: ```suggestion
```

> **To change the ownership, `nginx-ingress-controller` NodePort Service in `kube-system` namespace needs to be manually deleted. It then gets to be recreated by triggering the cluster upgrade. To minimize the downtime, please delegate cluster upgrades to your SE.**

Reviewer 2: ```suggestion
**Note for cluster upgrades:**

Please manually delete `nginx-ingress-controller` NodePort Service in `kube-system` namespace. Upgrading the cluster then recreates the NodePort Service and moves its management from kvm-operator to NGINX IC. To minimize downtime, please delegate cluster upgrades to your SE.**
```

@sslavic 
1. is this a required step? Or is it optional? 
1a If it is required (this is my impression), then I have suggestion in release notes. Pls check if makes it clearer.
1b If it's optional, then the instructions might be better placed in docs.
2. Also, this note applies to cluster upgrades only right? Would it be helpful to make that clearer?

## :zap:  Giant Swarm Release 12.1.0 for KVM :zap:

> **This release includes two significant improvements to NGINX Ingress Controller. It also includes a fix for Quay being a single point of failure by using Docker mirroring feature. This ensures availability of all images needed for node bootstrap, thus the cluster creation/scaling doesn’t depend on Quay availability anymore.**

Reviewer 2: ```suggestion
This release includes two **significant improvements to NGINX Ingress Controller**. It also includes a **fix for Quay being a single point of failure** by using Docker mirroring feature. This ensures availability of all images needed for node bootstrap, thus the cluster creation/scaling doesn’t depend on Quay availability anymore.
```

Formatting to bold only the important parts.

**The two NGINX Ingress Controller improvements:**

> - First, multiple NGINX Ingress Controllers per tenant cluster are now supported, enabling separation of internal vs external traffic, dev vs prod, and so on.

Reviewer 2: ```suggestion
- Multiple NGINX Ingress Controllers per tenant cluster are now supported, enabling separation of internal vs external traffic, dev vs prod, and so on.
```

> - Second, management of NGINX IC NodePort Service is moved from kvm-operator to NGINX IC App itself. This enables configurability of external traffic policy and lays the foundation for making NGINX IC App optional and not pre-installed in a future KVM platform release.

Reviewer 2: ```suggestion
- Management of NGINX IC NodePort Service is moved from kvm-operator to NGINX IC App itself. This enables configurability of external traffic policy and lays the foundation for making NGINX IC App optional and not pre-installed in a future KVM platform release.
```
